explorer/e2e/testFunctions.ts

```diff
@@ -522,4 +541,255 @@ export async function testClearAll(
     await page.locator("body").click();
   }
 }
+
+const getRowLocatorByAccess = (page: Page, access: string): Locator =>
+  page
+    .getByRole("row")
+    .filter({ has: page.getByRole("cell", { name: access }) })
+    .first()
+    .getByRole("cell")
+    .first()
+    .getByRole("link");
+
+// Backpages tests
+export async function testExportBackpage(
+  context: BrowserContext,
+  page: Page,
+  tab: TabDescription
+): Promise<void> {
+  if (tab.backpageExportButtons == null || tab.backpageAccessTags == null) {
+    // Fail if this test is ran on a tab without defined backpages
+    await expect(false);
+    return;
+  }
+  // Goto the specified tab
+  await page.goto(tab.url);
+  // Expect to find row with a granted status indicator
+  const grantedRowLocator = getRowLocatorByAccess(
+    page,
+    tab.backpageAccessTags.grantedShortName
+  );
+  await expect(grantedRowLocator).toBeVisible();
+  // Click into the selected row
+  await grantedRowLocator.dispatchEvent("click");
+  await expect(
+    page.getByText(tab.backpageExportButtons.detailsName)
+  ).toBeVisible();
+  // Click the "Export" tab
+  await page
+    .getByText(tab.backpageExportButtons.exportTabName, {
+      exact: true,
+    })
+    .click();
+  await expect(page).toHaveURL(tab.backpageExportButtons.exportUrlRegExp);
+  await expect(page.getByRole("checkbox").first()).toBeVisible();
+  const firstButtonLocator = page.getByRole("button", {
+    name: tab.backpageExportButtons.firstButtonName,
+  });
+  await expect(firstButtonLocator).toBeEnabled();
+  // Select all checkboxes on the pages
+  const checkboxLocators = await page.getByRole("checkbox").all();
+  for (const checkboxLocator of checkboxLocators) {
+    if (!(await checkboxLocator.isChecked())) {
+      await checkboxLocator.click();
+      await expect(checkboxLocator).toBeChecked();
+      await expect(checkboxLocator).toBeEnabled({ timeout: 10000 });
+    }
+  }
+  await expect(firstButtonLocator).toBeEnabled({ timeout: 10000 });
+  // Uncheck all checkboxes except one in each table, to reduce overhead
+  for (const tableLocator of await page.getByRole("table").all()) {
```

Comment:
Right now it does expect multiple tables, but if this will never be the case I can simplify. I think the way it's currently written, where it selects all checkboxes and then unchecks most of the ones that are in tables means that it doesn't make sense to reuse locators, but looking at it more there's definitely a better way to achieve what it currently does. I would make that change here, but since the export feature is currently disabled I don't know a good way to test it, so I'm not 100% sure what to do here.

Reply:
Let's go ahead and simplify this (to a single table) for now!